griptape/drivers/vector/mongodb_vector_store_driver.py

```diff
@@ -0,0 +1,85 @@
+from typing import Optional, Dict, Any, List
+from griptape.drivers import BaseVectorStoreDriver
+from pymongo import MongoClient
+from attr import define, field, Factory
+import logging
+
+
+@define
+class MongoDbAtlasVectorStoreDriver(BaseVectorStoreDriver):
+    connection_string: str = field(kw_only=True)
+    database_name: str = field(kw_only=True)
+    collection_name: str = field(kw_only=True)
+    client: MongoClient = field(init=False)
+    collection = field(init=False)
+
+    def __attrs_post_init__(self) -> None:
+        try:
+            self.client = MongoClient(self.connection_string)
+            self.collection = self.client[self.database_name][self.collection_name]
+            logging.info("Connected to MongoDB Atlas successfully!")
+        except Exception as e:
+            logging.error("Could not connect to MongoDB Atlas!")
+            logging.error(str(e))
+
+    def upsert_vector(
+            self,
+            vector: list[float],
+            vector_id: Optional[str] = None,
+            namespace: Optional[str] = None,
+            meta: Optional[dict] = None,
+            **kwargs
+    ) -> str:
+        try:
+            document = {"vector": vector, "meta": meta, "namespace": namespace}
+            if vector_id:
+                document["_id"] = vector_id
+            result = self.collection.replace_one({"_id": vector_id}, document, upsert=True)
+            return result.upserted_id or vector_id
+        except Exception as e:
```

Comment:
Do we want to swallow these exceptions or allow them to raise?

Reply:
Good catch! We should probably not use try-except here (and in other methods) at all since this is a low level driver.

Reply:
removed it !